moped-toolbox/geometry_prototype/create_prototype.sql

```diff
@@ -0,0 +1,141 @@
+    sidewalk_id integer default null,
+    sidewalk_name character varying default null,
+    geography geography('MULTILINESTRING') default null
+    ) inherits (features);
+
+create table drawn_points (
+    geography geography('MULTIPOINT') default null
+    ) inherits (features);
+
+create table drawn_lines (
+    geography geography('MULTILINESTRING') default null
+    ) inherits (features);
+
+create table component_feature_map (
+    id serial primary key,
+    component_id integer references components(id),
+    feature_id integer default null
+    );
+
+-- this could be (should be?) a CTE to the project_geography view, but this seems easier to read in a prototype
+create view uniform_features as (
+        select id, 'signals' as table, name, json_build_object('signal_id', signal_id) as attributes, geography
+        FROM signals
+    union all
+        select id, 'sidewalks' as table, name, json_build_object('sidewalk_id', sidewalk_id, 'sidewalk_name', sidewalk_name) as attributes, geography
+        FROM sidewalks
+    union all
+        select id, 'drawn_points' as table, name, null as attributes, geography from drawn_points
+    union all
+        select id, 'drawn_lines' as table, name, null as attributes, geography from drawn_lines
+);
+
+create view project_geography as (
+    select 
+      projects.id as project_id,
+      uniform_features.id as feature_id,
+      components.id as component_id,
+      projects.name as project_name,
+      layers.internal_table as table,
+      layers.reference_layer_primary_key_column as original_fk,
+      components.name as component_name,
+      uniform_features.name as feature_name,
+      uniform_features.attributes as attributes,
+      uniform_features.geography as geography
+    from projects
+    join components on (components.project_id = projects.id)
+    join component_types on (components.component_type_id = component_types.id)
+    join layers on (coalesce(components.layer_id_override, component_types.layer_id) = layers.id)
+    join component_feature_map on (components.id = component_feature_map.component_id)
+    join uniform_features on (component_feature_map.feature_id = uniform_features.id)
+);
+
+
+-- example project creation statements
+
+-- All inserts are specifying IDs, for ease of the prototype, but prostgres would normally be assigning those.
+-- Hardcoding an id into an insert is almost always not what you want.
+
+-- a project with a single component taken from the signals layer
+insert into projects (id, name) values (1, 'project_a');
+insert into components (id, project_id, component_type_id, name) values (1, 1, 1, 'A component (signal-phb) for project (project_a).');
+insert into signals (id, name, signal_id, geography) values (1, 'A new set of PHBs', 1001, ST_GeographyFromText(
+    'MULTIPOINT(-97.740556 30.274722, -97.725125 30.257440, -97.760225 30.286231)'
+    ));
+insert into component_feature_map (id, component_id, feature_id) values (1, 1, 1);
```

Comment:
yep, this is awesome and where the inheritance pays off. i'm assuming hasura will have no issue with this mutation but it will be good to test 🍿